Doc/library/asyncio-stream.rst

```diff
@@ -367,13 +366,177 @@ Stream
 
    Represents a Stream object that provides APIs to read and write data
    to the IO stream . It includes the API provided by :class:`StreamReader`
-   and :class:`StreamWriter`.
+   and :class:`StreamWriter`. It can also be used as :term:`asynchronous iterator`
+   where :meth:`readline` is used. It raises :exc:`StopAsyncIteration` when
+   :meth:`readline` returns empty data.
 
    Do not instantiate *Stream* objects directly; use API like :func:`connect`
    and :class:`StreamServer` instead.
 
    .. versionadded:: 3.8
 
+   .. attribute:: mode
+
+      Returns the mode of the stream which is a :class:`StreamMode` value.
+
+   .. method:: set_exception(exc)
+
+      Sets the exception to be raised when :meth:`read`, :meth:`readexactly` and
+      :meth:`readuntil` are called.
```

Comment:
Ouch.  @asvetlov Why is this a public API?

Reply:
Hard question.
It was public in the previous stream implementation.
We have two options:
1. Rename the method to `_set_exception()`.
2. Leave it as is but document that users should never touch it. Useless thing, I think (and a point of API misusage, like `Task.set_exception()`)

Reply:
I would like to rename the method to `_set_exception()` in 3.8.  If people come up with useful scenarios for that we can make it public in 3.9.  cc @ambv

Reply:
Preparing a PR.

Reply:
@tirkarthi please remove deprecated by #15762 methods and properties.
`asyncio.Stream` is a new class but it should support backward compatibility for the sake of asyncio subprocess subsystem.

There is no need to introduce deprecated-from-very-beginning API in docs.

Reply:
Thanks, removed `set_exception` and references to deprecated methods too in the docs as per the linked PR.